lib/ansible/utils/module_docs_fragments/kubernetes.py

```diff
@@ -0,0 +1,116 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+class ModuleDocFragment(object):
+
+    DOCUMENTATION = '''
+options:
+  state:
+    description:
+    - Determines if an object should be created, patched, or deleted. When set to
+      C(present), the object will be created, if it does not exist, or patched, if
+      parameter values differ from the existing object's attributes, and deleted,
+      if set to C(absent). A patch operation results in merging lists and updating
+      dictionaries, with lists being merged into a unique set of values. If a list
+      contains a dictionary with a I(name) or I(type) attribute, a strategic merge
+      is performed, where individual elements with a matching I(name_) or I(type)
```

Comment:
What is type in this context?  Since Kube has 3 different patch strategies (specified via content type), strategic merge patch may be versioned in the future and clients may have to customize it, it might actually be safer to not depend on strategic merge patch at all and instead make the content opaque.

Also, in the future Kube is considering having a "typed Patch" which *might* look something like:

```
kind: Patch
apiVersion: meta.k8s.io/v1
patchType: <mimeType>
patchData: <raw JSON>
target: <reference to an object, including Kind / Name / Namespace>
```

However, that could be further off.